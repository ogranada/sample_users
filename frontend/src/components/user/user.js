import React, { Component } from "react";
import { toTitleCase } from "components/utils.js";

import "./user.scss";

export class User extends Component {
  constructor(...args) {
    super(...args);
  }

  render() {
    const user = this.props.user;
    return (
      <div className="User">
        <img className="User-picture" src={user.picture} alt="" />
        <span className="User-name">
          {`${toTitleCase(user.firstname)} ${toTitleCase(user.lastname)}`}
        </span>
      </div>
    );
  }
}
