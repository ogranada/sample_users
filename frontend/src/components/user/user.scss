.User {
    display: flex;
    align-items: center;

    &-picture {
        border-radius: 50%;
    }

    &-name {
        padding-left: 1em;
    }

}