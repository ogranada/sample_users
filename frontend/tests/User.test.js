import React from "react";
import { shallow } from "enzyme";
import { User } from "components/user/user";

describe("User", () => {
  it("should render correctly", () => {
    const user = {
      firstname: "Oscar",
      lastname: "Granada",
      fullname: "Oscar Granada",
      picture: "http://fake.path"
    };
    const component = shallow(<User user={user} />);
    expect(component).toMatchSnapshot();
  });
});
