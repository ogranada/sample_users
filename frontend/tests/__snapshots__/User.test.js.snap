// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should render correctly 1`] = `
<div
  className="User"
>
  <img
    alt=""
    className="User-picture"
    src="http://fake.path"
  />
  <span
    className="User-name"
  >
    Oscar Granada
  </span>
</div>
`;
